instrumentation/trilogy/test/opentelemetry/instrumentation/trilogy/instrumentation_test.rb

```diff
@@ -376,13 +376,14 @@
       let(:config) { { propagator: 'vitess' } }
 
       it 'does inject context' do
-        sql = +'SELECT * from users where users.id = 1 and users.email = "test@test.com"'
-        original_sql = sql.dup
+        sql = 'SELECT * from users where users.id = 1 and users.email = "test@test.com"'.freeze # rubocop:disable Style/RedundantFreeze
+        propagator = double('propagator')
+        allow(propagator).to receive(:inject)
+        allow(client).to receive(:propagator).and_return(propagator)
         expect do
           client.query(sql)
         end.must_raise Trilogy::Error
-        encoded = Base64.strict_encode64("{\"uber-trace-id\":\"#{span.hex_trace_id}:#{span.hex_span_id}:0:1\"}")
-        _(sql).must_equal "/*VT_SPAN_CONTEXT=#{encoded}*/#{original_sql}"
+        expect(propagator).to have_received(:inject).with(sql, context: instance_of(OpenTelemetry::Context)).once
```

Comment:
How does this test ensure that a mutable string is what is passed in to the propagator?

How does it ensure what is returned is a frozen string that was potentially changed?

Reply:
Yeah, great point - I'm going to take another pass at specs.

Reply:
The tests ended up pretty gnarly on this 😳  - see e84b23d2fc13f7d7a2e5965066be871f5ede314e

Why gnarly? Based on my observation (i.e. not sourcediving, take it with a grain of salt, etc.), rspec-mocks appears to capture a _reference_ to the arguments passed when you use methods like `foo.receive(:inject).with(...)`. This means that if a method mutates an argument, rspec-mocks will only know about the argument's value at assertion time (vs. at call time). To get around this, I used `receive(:inject).and_wrap_original` to make assertions about frozenness etc. I plan to take another look at this tomorrow when my eyes are less glazed over 😛